pkg/app/middleware_test.go

```diff
@@ -0,0 +1,59 @@
+package app
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/pkg/thread"
+	"github.com/sirupsen/logrus"
+)
+
+func TestMiddleWare(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Middleware test")
+}
+
+var _ = Describe("Middleware test", func() {
+
+	l := logrus.New()
+	l.SetOutput(ioutil.Discard)
+
+	It("Healthcheck failure test", func() {
+		mHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+		failure := thread.New(l, "failed test", 100*time.Millisecond, func() {
+			time.Sleep(100 * time.Millisecond)
+		})
+		failure.Start()
+		failure.Stop()
+		req, err := http.NewRequest("GET", "/health", nil)
+		h1 := WithHealthMiddleware(mHandler, []*thread.Thread{failure}, l, 100*time.Millisecond)
+		Expect(err).NotTo(HaveOccurred())
+		rr := httptest.NewRecorder()
+		// wait more than monitored threshold
+		time.Sleep(300 * time.Millisecond)
+		h1.ServeHTTP(rr, req)
+		// Check the status code is what we expect.
+		Expect(rr.Code).To(Equal(http.StatusInternalServerError))
+	})
+
+	It("Healthcheck success test", func() {
+		mHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+		failure := thread.New(l, "failed test", 100*time.Millisecond, func() {
+			time.Sleep(10 * time.Millisecond)
+		})
+		failure.Start()
+		failure.Stop()
+		req, err := http.NewRequest("GET", "/health", nil)
+		h1 := WithHealthMiddleware(mHandler, []*thread.Thread{failure}, l, 600*time.Millisecond)
+		Expect(err).NotTo(HaveOccurred())
+		rr := httptest.NewRecorder()
+		h1.ServeHTTP(rr, req)
+		// Check the status code is what we expect.
+		Expect(rr.Code).To(Equal(http.StatusOK))
```

Comment:
The test should wait more than the monitored threshold.

Reply:
it shouldn't, this is happy case

Reply:
Still need to make sure that after the threshold you get OK

Reply:
don't understand. If i will get to threshold i will get an error. I want to check that if i didn't reach threshold i got OK